src/generator/pollers.ts

```diff
@@ -455,7 +434,35 @@ export async function generatePollersHelper(session: Session<CodeModel>): Promis
 	type pollingTrackerDelete struct {
 		pollingTrackerBase
 	}
-	
+  
+  // done queries the service to see if the operation has completed.
+  func (pt *pollingTrackerDelete) done(ctx context.Context, p azcore.Pipeline) bool {
```

Comment:
Can this be moved to `pollingTrackerBase` to avoid the duplication?  Probably need to add it to the interface.

Reply:
I tried that initially, but the methods called inside like updatePollingMethod() are implemented on the trackers and it marks an error when attempting to place it on the pollingTrackerBase.

Reply:
I could add similar methods on pollingTrackerBase

Reply:
Oh yes good point.  What about making `done()` a func instead that takes a `pollingTracker` interface as its third argument?

Reply:
Yes that could work, I'll do that then